src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -2253,21 +2258,37 @@ internal void GetNFIValues(NumberFormatInfo nfi)
         /// </remarks>
         internal static string AnsiToLower(string testString) => TextInfo.ToLowerAsciiInvariant(testString);
 
-        private int GetLocaleInfoCore(LocaleNumberData type) => GlobalizationMode.UseNls ?
-                                                                    NlsGetLocaleInfo(type) :
-                                                                    IcuGetLocaleInfo(type);
+        private int GetLocaleInfoCore(LocaleNumberData type)
+        {
+            if (GlobalizationMode.Invariant)
```

Comment:
>Invariant [](start = 34, length = 9)

should all checks inside the Core methods be just asserts? is it possible we can reach this code?

Reply:
No, this code should be unreachable as the Icu/Nls specific methods assert we're not in Invariant mode.

Reply:
could be this check added as a hint to the linker?

---
In reply to: [439672159](https://github.com/dotnet/runtime/pull/37832#discussion_r439672159) [](ancestors = 439672159)

Reply:
@marek-safar could you please add a little comment in such places to avoid anyone in the future remove this code again and tell why this code is here? Thanks.

Reply:
Extra comment added